tomviz/TomographyReconstruction.h

```diff
@@ -0,0 +1,40 @@
+/******************************************************************************
+ 
+ This source file is part of the tomviz project.
+ 
+ Copyright Kitware, Inc.
+ 
+ This source code is released under the New BSD License, (the "License").
+ 
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ 
+ ******************************************************************************/
+
+#ifndef tomvizTomographyReconstruction_h
+#define tomvizTomographyReconstruction_h
+
+#include "pqReaction.h"
+#include "vtkImageData.h"
+
+namespace tomviz
+{
+class DataSource;
+
+class TomographyReconstruction
+{
+    
+public:
+  TomographyReconstruction();
+  ~TomographyReconstruction();
+  static void weightedBackProjection3(vtkImageData *tiltSeries,vtkImageData *recon); //3D WBP recon
+  static void unweightedBackProjection2(float *sinogram, double *tiltAngles, float *recon,int numOfTilts, int numOfRays ); //2D WBP recon
+  static void tiltSeriesToSinogram(vtkImageData *tiltSeries, int, float* sinogram); //Extract sinograms from tilt series

```

Comment:
@mathturtle 
Let's say the dimension of a tilt series is (xDim,yDim,zDim). Because a sinogram is just a y-z plane of the tilt series, the pointer *sinogram points to a 1D (float) array of size yDim*zDim. 
Right now, the size of 3D reconstruction is always set to (xDim, yDim, yDim). For the 2D recon function "unweightedBackProjection2", the *recon points to a 1D array of size yDim*yDim. 
The parameter "numOfRays" is just yDim. Together with "numOfTilts", which is zDim, they specify the size of sinogram.

Because we have 2D reconstruction function now, there is no need to display sinograms in the UI. We can just display the 2D reconstructions.